src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,131 @@
+        sortData(playersScore, comparator);
+        saveDataToFile(playersScore, "stats.csv");
+
         // tutaj dodaj swoje rozwiązanie
         // użyj przekazanego scannera do wczytywania wartości
 
     }
+
+    private Comparator<Player> getPlayerComparator(Scanner scanner) {
+        Comparator<Player> sortComparator = getSortComparator(scanner);
+        Comparator<Player> resultComparator = getPositiveOrNegativeComparator(scanner, sortComparator);
+        return resultComparator;
+    }
+
+    private Comparator<Player> getPositiveOrNegativeComparator(Scanner scanner, Comparator<Player> sortComparator) {
+        System.out.printf("Sortować rosnąco czy malejąco? (%s - rosnąco, %s - malejąco)\n", ASCENDING, DESCENDING);
+        Comparator<Player> sortComparator1 = sortComparator;
+        int option = inputInteger(scanner);
+        switch (option) {
+            case ASCENDING:
+                sortComparator1 = sortComparator;
+                break;
+            case DESCENDING:
+                sortComparator1 = sortComparator.reversed();
+                break;
+            default:
+                System.out.println("błędne wybór spsoobu sortowania");
+                System.exit(0);
+        }
+        return sortComparator1;
+    }
+    
+    private void saveDataToFile(ArrayList<Player> playersScore, String fileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            for (Player player : playersScore) {
+                writer.write(player.toCsv());
+                writer.newLine();
+            }
+            System.out.println("Dane posortowano i zapisano do pliku " + fileName);
+        } catch (IOException e) {
+            System.out.println("błąd zapisu pliku");
+        }
+
+    }
+
+    private void sortData(ArrayList<Player> playersScore, Comparator<Player> comparator) {
+        Collections.sort(playersScore, comparator);
+    }
+
+    private Comparator<Player> getSortComparator(Scanner scanner) {
+        Comparator<Player> comparatotor = null;
+        System.out.printf("Po jakim parametrze posortować? (%s - %s, %s - %s, %s - %s)\n",
+                SORT_FIRSTNAME, FIRSTNAME, SORT_LASTNAME, LASTNAME, SORT_SCORE, SCORE);
+        int option = inputInteger(scanner);
+        switch (option) {
+            case SORT_FIRSTNAME:
+                comparatotor = new NameComparator();
+                break;
+            case SORT_LASTNAME:
+                comparatotor = new LastNameComparator();
+                break;
+            case SORT_SCORE:
+                comparatotor = new ScoreComparator();
+                break;
+            default:
+                System.out.println("błędne wybór spsoobu sortowania");
+                System.exit(0);
+        }
+        return comparatotor;
+    }
+
+    private ArrayList<Player> getPlayersScore(Scanner scanner) {
+        ArrayList<Player> players = new ArrayList<>();
```

Comment:
Jako typów używaj interfejsów zamiast ich implementacji, czyli tutaj List<Player>. Dzięki temu, jeśli okaże się, że lepsza będzie LinkedLista, albo jakaś sprawdzająca się w zastosowaniach wielowątkowych, to wystarczy wykonać zmianę w jednym miejscu.